core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java

```diff
@@ -174,6 +175,21 @@ private void acquireConnectionAndExecute(ClientRequestContext ctx, Endpoint endp
                                              HttpRequest req, DecodedHttpResponse res,
                                              ClientConnectionTimingsBuilder timingsBuilder,
                                              ProxyConfig proxyConfig) {
+        final InetSocketAddress remoteAddress = endpoint.toSocketAddress(-1);
```

Comment:
Notes:
- Double-checked that the remote host (not the proxy server) is used which seems correct
- Understood that although the name is `ipAddressFilter`, unix domain hosts may not provide an ip address but a path.